src/lib.rs

```diff
@@ -47,6 +47,7 @@
 #![doc(test(attr(allow(unused_extern_crates, unused_variables))))]
 #![doc(html_logo_url = "https://async.rs/images/logo--hero.svg")]
 #![recursion_limit = "1024"]
+#![feature(try_trait)]
```

Comment:
Depending on a nightly feature may be tricky for us because it means we wont' be able to run on stable for the 1.39 release. I'm thinking it may perhaps be better to scope `try_for_each` to `Result` for now, and mention this as a limitation to the lang team.

Reply:
Ok, but how does `std` work then? (i haven't had a moment to look closely atm)